### PR TITLE
Unattended mail relay

### LINKED ISSUE #1113: automate exim setup
create unattended variant of exim (mail relay) setup

### PR DESCRIPTION
**Pull request by mstormi**
Fixes #1113 

Signed-off-by: Markus Storm <markus.storm@gmx.net>

### CODE REVIEW

File: functions/packages.bash [hunk 1]
```diff
@@ -82,59 +82,66 @@ firemotd_setup() {
...
-  interfaces="$(dig +short "$HOSTNAME" | tr '\n' ';');127.0.0.1;::1"
+  # shellcheck disable=SC2154
+  interfaces="$(dig +short "$hostname" | tr '\n' ';')127.0.0.1;::1"
+  relaynets="$(dig +short "$hostname" | cut -d'.' -f1-3).0/24"
   introtext="We will guide you through the install of exim4 as the mail transfer agent on your system and configure it to relay mails through a public service such as Google gmail.\\n\\nThe values you need to enter after closing this window are documented here:\\n\\nhttps://github.com/openhab/openhabian/tree/master/docs/exim.md\\n\\nOpen that URL in a browser now. Your interface addresses are ${interfaces}.\\nYou will be able to repeat the whole installation if required by selecting the openHABian menu for MTA again."
   temp="$(mktemp "${TMPDIR:-/tmp}"/openhabian.XXXXX)"
 
   echo -n "$(timestamp) [openHABian] Beginning Mail Transfer Agent install and setup... "
-  if (whiptail --title "Mail Transfer Agent installation" --yes-button "Begin" --no-button "Cancel" --yesno "$introtext" 17 80); then echo "OK"; else echo "CANCELED"; return 0; fi
+  if [[ -z $INTERACTIVE ]]; then
```
Comment (ecdye) [inline functions/packages.bash:107]: Did you mean `[[ -n $INTERACTIVE ]]`?

File: functions/packages.bash [hunk 2]
```diff
@@ -82,59 +82,66 @@ firemotd_setup() {
...
+    if (whiptail --title "Mail Transfer Agent installation" --yes-button "Begin" --no-button "Cancel" --yesno "$introtext" 17 80); then echo "OK"; else echo "CANCELED"; return 0; fi
+  fi
 
-  echo "$(timestamp) [openHABian] Reconfiguring exim4-config... "
-  whiptail --title "exim4 Configuration" --msgbox "exim4-config is about to ask for information, please fill out each line." 7 80
-  if dpkg-reconfigure exim4-config; then echo "OK"; else echo "CANCELED"; return 0; fi
+  echo "$(timestamp) [openHABian] Configuring exim4-config... "
+  if [[ -z $INTERACTIVE ]]; then
```
Comment (ecdye) [inline functions/packages.bash:112]: Did you mean [[ -n $INTERACTIVE ]]?

**Review by ecdye — CHANGES_REQUESTED** [review]
Looks pretty good just a couple of things. Also, did you test to ensure that this sets up properly in both cases? You might also consider adding a integration with Amanda if all the proper config settings are provided to allow proper setup of both Exim and Amanda.

File: functions/packages.bash (hunk 1 above)
Comment (mstormi) [reply functions/packages.bash:107]: yes

File: functions/packages.bash (hunk 2 above)
Comment (mstormi) [reply functions/packages.bash:112]: turned this into a single if branch

**Comment by mstormi** [comment]
> You might also consider adding a integration with Amanda if all the proper config settings are provided to allow proper setup of both Exim and Amanda.

That of course was the context and idea of this. I checked Amanda code to find it's using "adminmail" variable (to send reports to) which is why I took that one as the admin's address rather than to query again/use another variable.
Frankly I'm not 100% sure I didn't overlook anything but that should be enough to integrate, or is there anything else related ?

**Review by ecdye — APPROVED** [review]
LGTM, merge when you would like.

**Pull request merged**